buffer_test.go

```diff
@@ -472,6 +472,38 @@ func mustInsert(t *testing.T, err error) {
 	}
 }
 
+func TestBrokenCursorBeforeFirstRecord(t *testing.T) {
+	var (
+		beforeFirstRecord1   = []byte{1,2,3}
+		beforeFirstRecord2   = []byte{1,2,3,4,5,6,7,8,9}
```

Comment:
Having several test cases would be great - negative and positive.

Reply:
tests for positive case exist, negative added